src/network/netdev/tunnel.c

```diff
@@ -161,6 +161,78 @@ static int netdev_gre_fill_message_create(NetDev *netdev, Link *link, sd_netlink
         return r;
 }
 
+static int netdev_erspan_fill_message_create(NetDev *netdev, Link *link, sd_netlink_message *m) {
+        uint32_t ikey, okey;
+        uint16_t iflags = 0;
+        uint16_t oflags = 0;
+        Tunnel *t;
+        int r;
+
+        assert(netdev);
+
+        t = ERSPAN(netdev);
+
+        assert(t);
+        assert(IN_SET(t->family, AF_INET, AF_UNSPEC));
+        assert(m);
+
+        r = sd_netlink_message_append_u32(m, IFLA_GRE_ERSPAN_INDEX, t->erspan_index);
+        if (r < 0)
+                return log_netdev_error_errno(netdev, r, "Could not append IFLA_GRE_ERSPAN_INDEX attribute: %m");
+
+        if (t->key != 0) {
+                ikey = okey = htobe32(t->key);
+                iflags |= GRE_KEY;
+                oflags |= GRE_KEY;
+        }
+
+        if (t->ikey != 0) {
+                ikey = htobe32(t->ikey);
+                iflags |= GRE_KEY;
+        }
+
+        if (t->okey != 0) {
+                okey = htobe32(t->okey);
+                oflags |= GRE_KEY;
+        }
+
+        if (t->sequence != -1) {
+                if (t->sequence) {
+                        iflags |= GRE_SEQ;
+                        oflags |= GRE_SEQ;
+                } else  {
+                        iflags &= ~GRE_SEQ;
+                        oflags &= ~GRE_SEQ;
+                }
```

Comment:
or actually, i think this would be best:

```c
if (t->sequence > 0) {
        iflags |= GRE_SEQ;
        oflags |= GRE_SEQ;
} else if (t->sequence == 0) {
        iflags &= ~GRE_SEQ;
        oflags &= ~GRE_SEQ;
}
```

that we we can avoid the nested if checks...